### ISSUE TITLE
CORS requests now fail due to `amz-sdk-invocation-id` header

### ISSUE THREAD

**Issue by monken** [aws/aws-sdk-js-v3#1376]
**Describe the bug**
With the recent release of gamma-1 CORS requests to the IoT data plane started failing because the `amz-sdk-invocation-id` is not a whitelisted CORS header

**SDK version number**
gamma-3

**Is the issue in the browser/Node.js/ReactNative?**
Browser

**To Reproduce (observed behavior)**

Access to fetch at 'https://xxxxxxxx-ats.iot.eu-west-1.amazonaws.com/things/thingId/shadow' from origin 'http://localhost:3000' has been blocked by CORS policy: Request header field amz-sdk-invocation-id is not allowed by Access-Control-Allow-Headers in preflight response.

**Expected behavior**
Only add the header when allowed by the remote CORS policy

**Comment by AllanFly120** [comment]
Hi @monken , we have filed an internal ticket to the service team to allow these new headers. I will provide more update here when I have more information. Meanwhile, you can pin the SDK to `1.0.0-gamma.2`.

**Comment by AllanFly120** [comment]
@monken IoT team is aware of this issue now and will deploy the fix soon.

**Comment by goodspeed3** [comment]
Have the same issue using Amplify Analytics
`    "@aws-amplify/analytics": {
      "version": "3.2.7",
      "resolved": "https://registry.npmjs.org/@aws-amplify/analytics/-/analytics-3.2.7.tgz",
      "integrity": "sha512-NfV9+CK6y1VRZaFjIZnSi2E+ycT8C3hDdJNN/Z5UFytdjXzVpHfaiw4juQv8yJAZ46FPDg8wJ5zhcQkLvLDHfA==",
      "requires": {
        "@aws-amplify/cache": "^3.1.23",
        "@aws-amplify/core": "^3.4.6",
        "@aws-sdk/client-firehose": "1.0.0-gamma.4",
        "@aws-sdk/client-kinesis": "1.0.0-gamma.4",
        "@aws-sdk/client-personalize-events": "1.0.0-gamma.4",
        "@aws-sdk/client-pinpoint": "1.0.0-gamma.4",
        "@aws-sdk/util-utf8-browser": "1.0.0-gamma.3",
        "uuid": "^3.2.1"
      }`

**Comment by mlabieniec** [comment]
Same deal with Kinesis Video SDK:

```
Access to fetch at 'https://r-xxx.kinesisvideo.us-east-2.amazonaws.com/v1/get-ice-server-config' from origin 'http://localhost:4200' has been blocked by CORS policy: Request header field amz-sdk-invocation-id is not allowed by Access-Control-Allow-Headers in preflight response.
zone.js:1152 POST https://r-xxx.kinesisvideo.us-east-2.amazonaws.com/v1/get-ice-server-config net::ERR_FAILED
(anonymous) @ zone.js:1152
...
```

**Comment by anantoghosh** [comment]
Same issue with Connect Client, downgraded to gamma 2 for now.

**Comment by AllanZhengYP** [comment]
@mlabieniec

Can you open a new issue so we can track it separately?
